pkg/handler/v2/networkdefaults/networkdefaults.go

```diff
@@ -147,6 +147,7 @@ func generateNetworkDefaults(provider kubermaticv1.ProviderType) apiv2.NetworkDe
 		},
 		ProxyMode:                resources.GetDefaultProxyMode(provider),
 		NodeLocalDNSCacheEnabled: resources.DefaultNodeLocalDNSCacheEnabled,
+		ClusterExposeStrategy:    defaulting.DefaultExposeStrategy,
```

Comment:
I think you need to override this in `overrideNetworkDefaultsByDefaultingTemplate`